include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp

```diff
@@ -252,19 +254,27 @@ class alignment_executor_two_way
         // Reset the get pointer.
         setg(std::ranges::begin(buffer), std::ranges::end(buffer));
 
+        using std::get;
+
         // Apply the alignment execution.
         size_t count = 0;
         size_t buffer_limit = in_avail();
-        for (; count < buffer_limit && !is_eof(); ++count, ++resource_it, ++gptr)
+        assert(buffer_limit >= chunk_size());
+
+        while (count < buffer_limit && !is_eof())
         {
-            auto && [tpl, idx] = *resource_it;
-            auto && [first_seq, second_seq] = tpl;
-            buffer_pointer write_to = gptr;
-            exec_handler.execute(kernel,
-                                 idx,
-                                 first_seq | views::all,
-                                 second_seq | views::all,
-                                 [write_to] (auto && res) { *write_to = std::move(res); });
+            for (auto && [sequence_pair, idx] : *resource_it)
+            {
+                buffer_pointer write_to = gptr;
```

Comment:
Why do you copy the pointer?
As far as I can see, the lambda copies it anyway?

Reply:
Sure thing. It was just from the old kernel. I can change that but will be modified soon by another PR anyway (see above). Would it be ok for you to leave this then as is?

Reply:
ok